### PR TITLE
(CDAP-13353) Moved HDFS Batch Sink from hydrator plugins as File Sink

### PR DESCRIPTION
**Pull request by bdmogal**
Code is basically copied from [HDFS Sink](https://github.com/caskdata/hydrator-plugins/blob/develop/hdfs-plugins/src/main/java/co/cask/hydrator/plugin/HDFSSink.java). The only change is to add a reference name, and not use ReferenceBatchSink and ReferencePluginConfig. 

Most other changes are in docs, widgets, and pom.xml

Corresponding PR to deprecate HDFS Sink: https://github.com/caskdata/hydrator-plugins/pull/787

Jira: [CDAP-13353](https://issues.cask.co/browse/CDAP-13353) 

Tested in a pipeline to write to GCS.

Will follow-up with a PR to push this to Cask Market

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,64 @@
...
+CDAP User Group and Development Discussions:
+
+* `cdap-user@googlegroups.com <https://groups.google.com/d/forum/cdap-user>`
+
+The *cdap-user* mailing list is primarily for users using the product to develop
+applications or building plugins for appplications. You can expect questions from 
+users, release announcements, and any other discussions that we think will be helpful 
+to the users.
+
+## IRC Channel
```
Comment (sreevatsanraman) [inline README.md]: We can remove IRC channel link

File: README.md [hunk 2]
```diff
@@ -1,2 +1,64 @@
...
+```    
+
+The build will create a .jar and .json file under the ``target`` directory.
+These files can be used to deploy your plugins.
+
+Deployment
+----------
+You can deploy your plugins using the CDAP CLI:
+
+    > load artifact <target/dynamic-spark-<version>.jar config-file <target/dynamic-spark-<version>.json>
```
Comment (sreevatsanraman) [inline README.md]: The artifact name needs to be fixed.

**Comment by sreevatsanraman** [comment]
Minor comments, LGTM otherwise. fix and merge

File: README.md (hunk 1 above)
Comment (bdmogal) [reply README.md]: Replaced with Slack workspace link

**Review by sreevatsanraman — APPROVED** [review]

**Comment by bdmogal** [comment]
There were some test failures. Fixed them by adding exclusions for hadoop dependencies:
image

**Pull request merged**